app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @post = Post.find(params[:post_id])
+    @comment.author_id = current_user.id
+    @comment.post_id = @post.id
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: 'Oops! An error has been occurred while creating the comment.'
+    end
+  end
```

Comment:
- [ ] There is an error raised when I try to add a new comment for a post, Please fix that.

![image](https://github.com/umair-rao/New-Blog-App/assets/68796232/efdcdc7a-a2c1-4faf-9244-1b556bc30284)